frontend/packages/console-plugin-sdk/src/typings/storage-class-params.ts

```diff
@@ -0,0 +1,35 @@
+import { Extension } from '@console/plugin-sdk';
+
+namespace ExtensionProperties {
+  export interface StorageClassProvisioner {
+    getStorageClassProvisioner: () => ExtensionSCProvisionerProp;
+  }
+}
+
+export interface StorageClassProvisioner
+  extends Extension<ExtensionProperties.StorageClassProvisioner> {
+  type: 'StorageClass/Provisioner';
+}
+
+export const isStorageClassProvisioner = (e: Extension): e is StorageClassProvisioner => {
+  return e.type === 'StorageClass/Provisioner';
+};
+
+export type ExtensionSCProvisionerProp = {
+  [key: string]: {
+    [key: string]: {
+      title: string;
+      provisioner: string;
+      parameters: {
+        [key: string]: {
+          name: string;
+          hintText: string;
+          value?: string;
+          visible?: () => boolean;
+          required?: boolean;
+          Component?: React.ComponentType<any>;
```

Comment:
shouldn't be any but
```suggestion
          Component?: React.ComponentType<ProvisionerProps>;
```

```
export type ProvisionerProps = {
  onParamChange: (id: string) => void;
};
```

You can also reuse the `ProvisionerProps` in `PoolResourceComponent`